File: CustomSources/src/main/java/com/github/eprendre/sources_java_demo/SourceEntry.java
package com.github.eprendre.sources_java_demo;

import com.github.eprendre.tingshu.sources.TingShu;
import kotlin.collections.CollectionsKt;

import java.util.List;

public final class SourceEntry {
    public static final SourceEntry INSTANCE;

    private SourceEntry() {
    }

    static {
        INSTANCE = new SourceEntry();
    }

    public static String getDesc() {
        return "测试源";
    }

    public static List<TingShu> getSources() {
        return CollectionsKt.listOf(M56TingShuJava.INSTANCE);
    }
}

File: CustomSources/src/main/kotlin/com/github/eprendre/sources_by_eprendre/SourceEntry.kt
package com.github.eprendre.sources_by_eprendre

import com.github.eprendre.tingshu.sources.TingShu

object SourceEntry {

    /**
     * 说明
     */
    @JvmStatic
    fun getDesc(): String {
        return "内置源"
    }

    /**
     * 返回此包下面的源
     */
    @JvmStatic
    fun getSources(): List<TingShu> {
        return listOf(
            HuanTingWang,
            JDLG,
            YouTuYueDu,
            YunTuYouSheng,
            WoTingPingShu,
            KuWo,
            KouDaiWeiKeTang,
            BoKanYouSheng,
            ZhongBanYouSheng,
            CCTV,
            YouShengXiaoShuoBa,
            ShengBoFM,
            IFish,
            TingChina,
            BiliBili,
            JingTing,
            XinMo,
            TingShu74,
            TingShuBao,
            //            MaLaTIngShu
//                YouYuKu,
//                HaiDao,
            SouGou,
            Mgting
        )
    }
}